third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -458,6 +474,21 @@ func resourceComputeInstanceGroupManagerRead(d *schema.ResourceData, meta interf
 		return err
 	}
 
+	operation := d.Get("operation").(string)
+	if operation != "" {
+		log.Printf("[DEBUG] in progress operation detected at %v, attempting to resume", operation)
+		zone, _ := getZone(d, config)
+		op := &computeBeta.Operation{
+			Name: operation,
+			Zone: zone,
+		}
+		d.Set("operation", "")
+		err = computeOperationWaitTime(config, op, project, "Creating InstanceGroupManager", d.Timeout(schema.TimeoutCreate))
```

Comment:
I think that this may cause issues. If terraform is cancelled while waiting on an operation that fails the user gets into a bad state that will require manual fixing.

This happened to me when manually testing: I made a request to create an IGM that failed with a 400 bad request at some point during the operation, but I cancelled before it returned that.

When resuming the operation `computeOperationWaitTime` returns an error because the operation is `DONE` but contains errors, preventing any plan or applies from succeeding

Reply:
I see. The solution would be to remove the resource from state in this scenario?